src/java.base/share/classes/java/lang/ThreadDeath.java

```diff
@@ -26,26 +26,19 @@
 package java.lang;
 
 /**
- * An instance of {@code ThreadDeath} is thrown in the victim thread
- * when the (deprecated) {@link Thread#stop()} method is invoked.
+ * This error is no longer thrown.
  *
- * <p>An application should catch instances of this class only if it
- * must clean up after being terminated asynchronously.  If
- * {@code ThreadDeath} is caught by a method, it is important that it
- * be rethrown so that the thread actually dies.
- *
- * <p>The {@linkplain ThreadGroup#uncaughtException top-level error
- * handler} does not print out a message if {@code ThreadDeath} is
- * never caught.
- *
- * <p>The class {@code ThreadDeath} is specifically a subclass of
- * {@code Error} rather than {@code Exception}, even though it is a
- * "normal occurrence", because many applications catch all
- * occurrences of {@code Exception} and then discard the exception.
+ * @deprecated {@link Thread#stop()} was originally specified to "stop" a
+ *      thread by causing it to throw a {@code ThreadDeath}. It was inherently
+ *      unsafe and deprecated in an early JDK release. The ability to "stop" a
+ *      thread with {@code Thread.stop} has been removed and the {@code
+ *      Thread.stop} method changed to throw an exception. Consequently,
+ *      {@code ThreadDeath} is also deprecated and may be deprecated
+ *      for removal in a future release.
  *
  * @since   1.0
  */
-
+@Deprecated(since="20")
 public class ThreadDeath extends Error {
```

Comment:
Actually, `ThreadDeath` is thrown manually by **JShell** instrumentation code since [JDK‑8289613] ([GH‑10166])[^1].

[JDK‑8289613]: https://bugs.openjdk.org/browse/JDK-8289613
[GH‑10166]: https://github.com/openjdk/jdk/pull/10166
[^1]: https://github.com/openjdk/jdk/blob/00befddd7ce97d324250807824469daaa9434eef/src/jdk.jshell/share/classes/jdk/jshell/execution/LocalExecutionControl.java#L104-L107

Reply:
> Actually, `ThreadDeath` is thrown manually by **JShell** instrumentation code since JDK‑8289613.

That's right but it will likely have to use something if this error is terminally deprecated and eventually removed.